eo-runtime/src/test/java/org/eolang/io/EOstdinTest.java

```diff
@@ -0,0 +1,139 @@
+import org.junit.jupiter.api.Assertions;
+
+/**
+ * Test case for {@link EOstdin}.
+ *
+ * @since 0.1
+ */
+public final class EOstdinTest {
+
+    private static final InputStream DEFAULT_STDIN = System.in;
+ 
+    @AfterAll
+    public static void rollbackChangesToStdin() {
+        System.setIn(DEFAULT_STDIN);
+    }
+ 
+    @Test
+    public void nextLineOneLineTest() throws Exception {
+        String expectedResult = "this is a test input!";
+        mockSystemIn(expectedResult + "\n");
+
+        final Phi nextLineCopy = new PhMethod(new PhCopy(new EOstdin(new PhEta())), "nextLine");
+        final String gatheredInput = new Datarized(nextLineCopy).take(String.class);
+        MatcherAssert.assertThat(
+            gatheredInput,
+            Matchers.equalTo(expectedResult)
+        );
+    }
+
+    @Test
+    public void nextLineMultiLineTest() throws Exception {
+        String expectedResult = "this is a test input!";
+        String input = expectedResult + "\nanother line\nyet another line";
+        mockSystemIn(input);
+
+        final Phi nextLineCopy = new PhMethod(new PhCopy(new EOstdin(new PhEta())), "nextLine");
+        final String gatheredInput = new Datarized(nextLineCopy).take(String.class);
+        MatcherAssert.assertThat(
+            gatheredInput,
+            Matchers.equalTo(expectedResult)
+        );
+    }
+
+    @Test
+    public void nextLineEmptyTest() throws Exception {
+        String input = "";
+        mockSystemIn(input);
+        final Phi result = new PhMethod(new PhCopy(new EOstdin(new PhEta())), "nextLine");
+        MatcherAssert.assertThat(
+            new Datarized(
+                result.attr("msg").get()
+            ).take(String.class),
+            Matchers.equalTo("There is no line in the standard input stream to consume")
+        );
+        Assertions.assertThrows(org.eolang.phi.Attr.Exception.class, () -> {
+            new Datarized(result).take(String.class);
+        });
+    }
+
+    @Test
+    public void stdinOneLineTest() throws Exception {
+        String input = "this is a test input!\n";
+        mockSystemIn(input);
+
+        final Phi stdinCopy = new PhCopy(new EOstdin(new PhEta()));
+        final String gatheredInput = new Datarized(stdinCopy).take(String.class);
+        MatcherAssert.assertThat(
+            gatheredInput,
+            Matchers.equalTo(input)
+        );
+    }
+
+    @Test
+    public void stdinMultiLineTest() throws Exception {
+        String input = "this is a test input!\nanother line\nyet another line";
+        mockSystemIn(input);
+
+        final Phi stdinCopy = new PhCopy(new EOstdin(new PhEta()));
+        final String gatheredInput = new Datarized(stdinCopy).take(String.class);
+        MatcherAssert.assertThat(
+            gatheredInput,
+            Matchers.equalTo(input)
+        );
+    }
+
+    @Test
+    public void stdinEmptyTest() throws Exception {
+        String input = "";
+        mockSystemIn(input);
```

Comment:
@nlchar this is pretty dirty. You set it, but never reset back. What if this conflicts with other tests?

Reply:
@yegor256 Yegor, I reset System.in in the `@AfterAll rollbackChangesToStdin method of this test class`